tests/harvest_api.py

```diff
@@ -777,5 +773,129 @@ def delete_project_user(project_id, project_user_id):
+        client_1 = create_client()
+        cls._teardown_delete["clients"].append({"id": client_1["id"]})
+
+        # Contacts
+        contact_1 = create_contact(client_1['id'])
+        cls._teardown_delete["contacts"].append({"id": contact_1["id"]})
+
+        # Projects
+        if itter < 2: # Protect against project max (2)
+            project_1 = create_project(client_1['id'])
+            cls._teardown_delete["projects"].append({"id": project_1["id"]})
+        project_user_1 = create_project_user(project_1['id'], get_random("users"))
+        cls._teardown_delete["project_users"].append({"id": project_user_1["id"]})
+        
+        # Tasks
+        task_1 = create_task()
+        cls._teardown_delete["tasks"].append({"id": task_1["id"]})
+        project_task_1 = create_project_task(project_1['id'], task_1['id'])
+        cls._teardown_delete["project_tasks"].append({"id": project_task_1["id"]})
+
+        # Users ( This is necessary to ensure user_projects data exists)
+        user_1 = update_user(get_random('users'))
+
+        # Roles
+        role_1 = create_role()
+        cls._teardown_delete["roles"].append({"id": role_1["id"]})
+        user_role_1 = update_role(role_1['id'], [get_random("users")])
+        cls._teardown_delete["user_roles"].append({"id": user_role_1["id"]})
+        
+        # Estimates
+        estimate_1 = create_estimate(client_1['id'])
+        cls._teardown_delete["estimates"].append({"id": estimate_1['id']})
+        estimate_message_1 = create_estimate_message(estimate_1['id'])
+        cls._teardown_delete["estimate_messages"].append({"id": estimate_message_1['id']})
+        estimate_item_category_1 = create_estimate_item_category()
+        cls._teardown_delete["estimate_item_categories"].append({"id": estimate_item_category_1['id']})
+
+        # Invoices
+        invoice_1 = create_invoice(client_id=client_1['id'], project_id=project_1['id'])
+        cls._teardown_delete["invoices"].append({"id": invoice_1["id"]})
+        invoice_message_1 = create_invoice_message(invoice_1['id'])
+        cls._teardown_delete["invoice_messages"].append({"id": invoice_message_1["id"]})
+        invoice_payment_1 = create_invoice_payment(invoice_1['id'])
+        cls._teardown_delete["invoice_payments"].append({"id": invoice_payment_1["id"]})
+        invoice_item_category_1 = create_invoice_item_category()
+        cls._teardown_delete["invoice_item_categories"].append({"id": invoice_item_category_1["id"]})
+
+        # Time Entries
+        time_entry_1 = create_time_entry(project_1['id'], task_1['id'])
+        cls._teardown_delete["time_entries"].append({"id": time_entry_1["id"]})
+
+        # Expesnes
+        expense_category_1 = create_expense_category()
+        cls._teardown_delete["expense_categories"].append({"id": expense_category_1['id']})
+        expense_1 = create_expense(project_1['id'])
+        cls._teardown_delete["expenses"].append({"id": expense_1['id']})
+
+def tear_cown_class(cls):
```

Comment:
```suggestion
def tear_down_class(cls):
```

Reply:
Also, add function comments and benefits of this method.

Reply:
Added function comments to explain `test_down_class`.